lib/server.js

```diff
@@ -131,6 +132,18 @@ Server.prototype.refreshFiles = function () {
 
 Server.prototype._start = function (config, launcher, preprocess, fileList,
                                     capturedBrowsers, executor, done) {
+  const karmaJsPath = path.join(__dirname, '/../static/karma.js')
+  const contextJsPath = path.join(__dirname, '/../static/context.js')
+  // Check if the static files haven't been compiled
+  if (!(fs.existsSync(karmaJsPath) && fs.existsSync(contextJsPath))) {
+    this.log.info('Front-end scripts not present. Compiling...')
+    // Browserify the files asynchronously
+    var bundler = browserify(path.join(__dirname, '/../client/main.js'))
+    bundler.bundle().pipe(fs.createWriteStream(karmaJsPath)).once('finish', () => {
+      bundler = browserify(path.join(__dirname, '/../context/main.js'))
+      bundler.bundle().pipe(fs.createWriteStream(contextJsPath))
```

Comment:
This creates a race condition, if the bundle is not generated fast enough, it will be missing when trying to be served. Need to make sure things are blocked to move on until it is finished.